### DISCUSSION TITLE
AngularJS não reconhece modificações de estado

### DISCUSSION THREAD

**Discussion by pcfmello** [frontendbr/forum#467] [Perguntas e Respostas]
Eu tenho um problema aqui na empresa num sistema que é o seguinte: Temos a tarefa de criar um editor de texto com imagens etc usando contenteditable etc, na vdd um gerador de catálogos digitais, e ai temos um método de salvamento que salvamos todo o html e css (meta) do mesmo porém, como precisamos formatar o texto etc, criamos os botões de formatação utilizando o execCommand do javascript. Eis o problema agora… O AngularJS não reconhece as modificações que são feitas através do comando citado anteriormente e por esse motivo não conseguimos salvar essas modificações, pois o sistema está projetado para salvar automaticamente cada mudança do meta. Alguém já teve esse problema? Como resolveu? Se não, alguém poderia me dar o caminho para que eu possa resolver isso?
Desde já agradeço a atenção de vocês...

**Comment by joao-alberto** [comment]
Amigo, da uma olhada [nisso aqui](https://github.com/fraywing/textAngular/), talvez você não use ele, mas pode dar uma olhada no código pra ver como funciona. ✌️

**Comment by pcfmello** [comment]
@psicotropicos , vou olhar sim... muito obrigado por enquanto :)

**Comment by marcoblos** [comment]
@pcfmello existem alguns "problemas" que na real não são "problemas de verdade" quando se usa AngularJS (não sei se angular 2 funciona dessa maneira que vou falar, nunca usei, estou falando por Angular 1.x, isso SE eu não estiver errado hahaha).
Esses problemas são coisas do tipo: "Se você utiliza Angular, então, você não deve utilizar 'Javascript normal', utilize as 'features' do Angular que nós cuidamos da parte de sincronização de DOM pra você".
Podemos ver isso através das diferentes features do angular:
- Com angular não devemos utilizar ``timeout`, devemos utilizar `$timeout`
- Com angular não devemos utilizar `interval` devemos utilizar `$interval`
- ...

Existem diversas diferenças no angular, mas basicamente o que acaba acontecendo é que se tu utilizar `timeout` no lugar de `$timeout`, existe a possibilidade, dependendo de como for tua implementação, da _view_ não ser atualizada. Nesses casos, TALVEZ forçar a atualização do escopo resolva. Utilizando o `$scope.$apply`.

Repetindo: Existe a chance de eu ter dito algo que não representa exatamente a realidade do Angular ou tua necessidade,, mas vale a tentativa.
Espero que ajude =)
